### PR TITLE
Test java 17 in CI

### PR DESCRIPTION
**Pull request by bantonsson**
# What Does This Do

Adds Java 17 as one of the tested versions in CI.

# Motivation

Test on the latest LTS release

# Additional Notes

The RMI instrumentation won't work on Java 17 right now since it inherits across modules in an unsupported way. This needs to be fixed.

### CODE REVIEW

File: dd-java-agent/instrumentation/elasticsearch/rest-5/rest-5.gradle [hunk 1]
```diff
@@ -1,6 +1,7 @@
 // Set properties before any plugins get loaded
 ext {
   minJavaVersionForTests = JavaVersion.VERSION_1_8
+  maxJavaVersionForTests = JavaVersion.VERSION_15
```
Comment (devinsba) [inline dd-java-agent/instrumentation/elasticsearch/rest-5/rest-5.gradle:4]: I'm guessing elasticsearch server doesn't start with Java 17 too? Might be good to have comments for all these things

**Review by devinsba — APPROVED** [review]
It's less changes than I expected. We can get around the cassandra/elasticsearch issues if someone can figure out getting docker working on our circleci runners

File: dd-java-agent/instrumentation/elasticsearch/rest-5/rest-5.gradle (hunk 1 above)
Comment (bantonsson) [reply dd-java-agent/instrumentation/elasticsearch/rest-5/rest-5.gradle:4]: Yes, you are right. I should go through them and add comments, and a consistent `TODO Java 17 ...` where applicable.

**Comment by bantonsson** [comment]
@devinsba Since our CircleCI runners are `docker` you can't run another `docker` inside that `docker`, as far as I can tell, but you can have other `docker` containers running at the same time (like we do for `memcached`, `rabbitmq`, `mongo`, et.c.) I don't know how much that weighs down the machine though, since they run for the whole duration of the testing.

**Pull request merged**